apps/users/adapters.py

```diff
@@ -54,13 +54,13 @@ def get_email_confirmation_url(self, request, emailconfirmation):
     def send_mail(self, template_prefix, email, context):
         return UserAccountEmail.send(email, template_name=template_prefix, **context)
 
-    def get_email_confirmation_redirect_url(self, request):
-        if "next" in request.GET and url_has_allowed_host_and_scheme(
-            request.GET["next"], allowed_hosts=None
+    def get_email_verification_redirect_url(self, email_address):
+        if "next" in self.request.GET and url_has_allowed_host_and_scheme(
+            self.request.GET["next"], allowed_hosts=None
         ):
-            return request.GET["next"]
+            return self.request.GET["next"]
         else:
-            return super().get_email_confirmation_redirect_url(request)
+            return super().get_email_verification_redirect_url(email_address)
```

Comment:
@goapunk this was changed in allauth codebase. Don't know why email_address is there, as I don't see it being used in the super's method either, but this is how it's constructed.